### ISSUE TITLE
Exception in activity function fails orchestrator with non-deterministic error

### ISSUE THREAD

**Issue by anthonychu** [Azure/azure-functions-durable-js#125]
**Describe the bug**
An exception in an activity results in a non-deterministic error.

**Investigative information**

- Durable Functions extension version: 2.0.0
- durable-functions npm module version: 1.3.1
- Language (JavaScript/TypeScript) and version: JS
- Node.js version: 10.16.3

**To Reproduce**

1. Clone this repo: https://github.com/anthonychu/test-durable-js
1. ```bash
    npm i
    func extensions install
    cp local.settings.SAMPLE.json local.settings.json
    # Add storage connection string to local.settings.json
    func start
    ```
1. Call `http://localhost:7071/api/orchestrators/OrchestratorWithBadActivity` to start the orchestrator that calls an activity that throws an exception.
1. Get the `statusQueryGetUri` in the response and use it to check the status. I think it should show the actual error, but instead it shows a non-deterministic error:
    ```json
    {
      "name": "OrchestratorWithBadActivity",
      "instanceId": "4db326c0a2d34fdbad47cf134e6ddf1c",
      "runtimeStatus": "Failed",
      "input": null,
      "customStatus": null,
      "output": "Non-Deterministic workflow detected: TaskScheduledEvent: 0 TaskScheduled BadActivity",
      "createdTime": "2019-11-20T22:27:37Z",
      "lastUpdatedTime": "2019-11-20T22:27:38Z"
    }
    ```

@ConnorMcMahon @cgillum

**Comment by ConnorMcMahon** [comment]
@antempus for reference

In order to fully fix this issue, apps will have to meet the following conditions.

The use of Durable JS 1.3.3 or higher
AND
(The use of the Durable Functions extension v1.8.5 or greater if using Durable 1.x,
OR 
The use of Durable Functions extension 2.2.0 or greater if using Durable 2.x
)

The Durable JS update should be expected next week, and the 1.8.5 and 2.2.0 releases late this month  or early next month. 

If using extension bundles, it will likely be a few weeks before 1.8.5 becomes the default extension version used.

**Comment by ConnorMcMahon** [comment]
The fix going into v1.8.5 of the extension is here: https://github.com/Azure/azure-functions-durable-extension/pull/1171.

**Comment by antempus** [comment]
Awesome! Thanks for the update.

**Comment by anthonychu** [comment]
Looks like this is fixed with 1.8.5. Currently requires `func extensions install` that version. Bundles update with this version should be available over the next few weeks, tracked here: https://github.com/Azure/azure-functions-extension-bundles/issues/13

Feel free to reopen if it's still not addressed.
